adapters/taboola/taboola.go

```diff
@@ -0,0 +1,175 @@
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+	for _, seatBid := range response.SeatBid {
+		for i := range seatBid.Bid {
+			bidType, err := getMediaType(seatBid.Bid[i].ImpID, request.Imp)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			b := &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: bidType,
+			}
+			bidResponse.Bids = append(bidResponse.Bids, b)
+		}
+	}
+	return bidResponse, errs
+}
+
+func createTaboolaRequest(request *openrtb2.BidRequest) (taboolaRequest *openrtb2.BidRequest, errors []error) {
+	modifiedRequest := *request
+	var errs []error
+
+	var taboolaExt openrtb_ext.ImpExtTaboola
+	for i := 0; i < len(modifiedRequest.Imp); i++ {
+		imp := modifiedRequest.Imp[i]
+
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if err := json.Unmarshal(bidderExt.Bidder, &taboolaExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if taboolaExt.TagId != "" {
+			imp.TagID = taboolaExt.TagId
+			modifiedRequest.Imp[i] = imp
+		}
+		if taboolaExt.BidFloor != 0 {
+			imp.BidFloor = taboolaExt.BidFloor
+			modifiedRequest.Imp[i] = imp
+		}
+	}
+
+	publisher := &openrtb2.Publisher{
+		ID: taboolaExt.PublisherId,
+	}
+
+	if modifiedRequest.Site == nil {
+		newSite := &openrtb2.Site{
+			ID:        taboolaExt.PublisherId,
+			Name:      taboolaExt.PublisherId,
+			Domain:    evaluateDomain(taboolaExt.PublisherDomain, request),
+			Publisher: publisher,
+		}
+		modifiedRequest.Site = newSite
+	} else {
+		modifiedSite := *modifiedRequest.Site
+		modifiedSite.Publisher = publisher
+		modifiedSite.ID = taboolaExt.PublisherId
+		modifiedSite.Name = taboolaExt.PublisherId
+		modifiedSite.Domain = evaluateDomain(taboolaExt.PublisherDomain, request)
+		modifiedRequest.Site = &modifiedSite
+	}
+
+	if taboolaExt.BCat != nil {
+		modifiedRequest.BCat = taboolaExt.BCat
+	}
+
+	if taboolaExt.BAdv != nil {
+		modifiedRequest.BAdv = taboolaExt.BAdv
+	}
+
+	return &modifiedRequest, errs
+}
+
+func getMediaType(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	for _, imp := range imps {
+		if imp.ID == impID {
+			if imp.Native != nil {
+				return openrtb_ext.BidTypeNative, nil
```

Comment:
Could you cover this line of code with a test case?

Reply:
@AlexBVolcy 
In the current phase we support only banner and not native, we plan to support native very soon but not in this PR. 
The condition handling in the function was a preparation. However, I removed the line and checking now for banner impressions only to prevent discrepancy. 
In the future when we are ready will create a dedicated FB for this. Thanks!